### PR TITLE
MM-51786: Sentry crash while installing a blocked plugin

### PR DESCRIPTION
**Pull request by agnivade**
We would return "nil, nil" if a plugin was on the blocklist.
This would cause a nil dereference panic while trying to add
struct fields from manifest to the audit logs.

To avoid it, we return an error explicitly and ignore the
error ids to unnecessarily have it log an error.

https://mattermost.atlassian.net/browse/MM-51786

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/plugin_install.go
Comment (crspeller) [inline server/channels/app/plugin_install.go:1]: There there is a second point in that function where we return `nil, nil`. Line 371. Maybe we should fix it there too?

File: server/channels/app/plugin_install_test.go [hunk 1]
```diff
@@ -172,11 +172,9 @@ func TestInstallPluginLocally(t *testing.T) {
...
 		cleanExistingBundles(t, th)
 
-		manifest, appErr := installPlugin(t, th, "playbooks", "0.0.1", installPluginLocallyAlways)
-		require.Nil(t, appErr)
-		require.Nil(t, manifest)
-
-		assertBundleInfoManifests(t, th, []*model.Manifest{})
+		_, appErr := installPlugin(t, th, "playbooks", "0.0.1", installPluginLocallyAlways)
+		require.NotNil(t, appErr)
+		require.Equal(t, "app.plugin.blocked.app_error", appErr.Id)
```
Comment (crspeller) [inline server/channels/app/plugin_install_test.go:177]: Why did you remove the assertion that it doesn't load it?

File: server/i18n/en.json [hunk 2]
```diff
@@ -5927,6 +5927,10 @@
     "id": "app.oauth.update_app.updating.app_error",
     "translation": "We encountered an error updating the app."
   },
+  {
+    "id": "app.plugin.blocked.app_error",
+    "translation": "Plugin {{.Id}} cannot be installed since it's already packaged with Mattermost."
```
Comment (crspeller) [inline server/i18n/en.json:5932]: I did make this generic enough that we could use it for other things. Maybe the error message should be generic too?

**Review by crspeller — CHANGES_REQUESTED** [review]

File: server/channels/app/plugin_install_test.go (hunk 1 above)
Comment (agnivade) [reply server/channels/app/plugin_install_test.go:177]: Ah my bad

File: server/i18n/en.json (hunk 2 above)
Comment (agnivade) [reply server/i18n/en.json:5932]: 0/5, but the debug log already mentions Skipping installation of plugin since plugin is on blocklist. Do you still want me to change it?

**Comment by agnivade** [comment]
@crspeller - PTAL

File: server/i18n/en.json (hunk 2 above)
Comment (crspeller) [reply server/i18n/en.json:5932]: Doesn't really matter. If we use it's genetic form we can change the message.

**Review by crspeller — APPROVED** [review]

File: server/i18n/en.json (hunk 2 above)
Comment (agnivade) [reply server/i18n/en.json:5932]: Any suggestions on what the message should be? I want to include some reason as well, because just reading that the plugin is blocked will make users ask why.

Comment (crspeller) [reply server/i18n/en.json:5936]: `Plugin {{.Id}} is on the block list. Some plugins are blocked because they are built into this version of Mattermost`

**Review by crspeller — APPROVED** [review]

**Comment by agnivade** [comment]
Asking for more reviews since Jose is out.

File: server/i18n/en.json [hunk 3]
```diff
@@ -6063,6 +6067,10 @@
     "id": "app.plugin.signature_decode.app_error",
     "translation": "Unable to decode base64 signature."
   },
+  {
+    "id": "app.plugin.skip_installation.app_error",
+    "translation": "Skipping installation of plugin {{.Id}} since existing version is newer."
```
Comment (trilopin) [inline server/i18n/en.json:6072]: 1/5 on the suggested wording.

If we try to install the same version, the error will be raised telling us that the existing one is newer (but it's the same). It would be nice to have a more accurate message here and in the mlog line.

```suggestion
    "translation": "Skipping installation of plugin {{.Id}} since existing version is equal or newer."
```

**Review by trilopin — APPROVED** [review]
LGTM

left an optional comment of log/error-message accuracy
